@here/harp-mapview/lib/VisibleTileSet.ts

```diff
@@ -246,78 +313,96 @@ export class VisibleTileSet {
+                if (dataSource.shouldRender(displayZoomLevel, tileKey)) {
                     visibleTiles.push(tileEntry);
                 }
 
-                tilingScheme.getSubTileKeys(tileEntry.tileKey).forEach(childTileKey => {
-                    const intersectsFrustum = tileFrustumIntersectionCache.get(
-                        childTileKey.mortonCode()
-                    );
-
-                    let subTileArea = 0;
-
-                    if (intersectsFrustum === undefined) {
-                        const geoBox = tilingScheme.getGeoBox(childTileKey);
-                        this.options.projection.projectBox(geoBox, tileBounds);
-                        tileBounds.min.sub(worldCenter);
-                        tileBounds.max.sub(worldCenter);
-
-                        if (
-                            (!this.options.extendedFrustumCulling ||
-                                this.m_mapTileCuller.frustumIntersectsTileBox(tileBounds)) &&
-                            this.m_frustum.intersectsBox(tileBounds)
-                        ) {
-                            const contour = [
-                                new THREE.Vector3(
-                                    tileBounds.min.x,
-                                    tileBounds.min.y,
-                                    0
-                                ).applyMatrix4(this.m_viewProjectionMatrix),
-                                new THREE.Vector3(
-                                    tileBounds.max.x,
-                                    tileBounds.min.y,
-                                    0
-                                ).applyMatrix4(this.m_viewProjectionMatrix),
-                                new THREE.Vector3(
-                                    tileBounds.max.x,
-                                    tileBounds.max.y,
-                                    0
-                                ).applyMatrix4(this.m_viewProjectionMatrix),
-                                new THREE.Vector3(
-                                    tileBounds.min.x,
-                                    tileBounds.max.y,
-                                    0
-                                ).applyMatrix4(this.m_viewProjectionMatrix)
-                            ];
-
-                            contour.push(contour[0]);
-
-                            const n = contour.length;
-
-                            for (let p = n - 1, q = 0; q < n; p = q++) {
-                                subTileArea +=
-                                    contour[p].x * contour[q].y - contour[q].x * contour[p].y;
+                tilingScheme.getSubTileKeys(tileKey).forEach(childTileKey => {
+                    // Here we attempt to find unique Tiles with +/- 5 offsets (which means
+                    // there is a total of 11 possible tiles rendered)
+                    const longitudeOffset = Math.round(worldGeoPoint.longitudeInDegrees / 360);
```

Comment:
Intuition says, that we have search for different offsets only at root level.
If you happen to hit different offset root tile, all the children will be selected automatically.

So i think that current way of searching is correct, but have to be parametrized with offset and wrapped in something like this:
And just call it like this:
```
newGetTiles(...) {
   for (offsetCandidate in range[-5,5]) {
      oldGetTiles(..., offsetCandidate)
  }
}
```

Just thinking, i may be wrong.

Reply:
Good point, there is no need to test so many offsets I will fix.

Reply:
Good point, there is no need to test so many offsets I will fix.

Reply:
Still open